contracts/src/v0.8/automation/dev/v2_3/AutomationRegistryLogicB2_3.sol

```diff
@@ -35,6 +37,37 @@ contract AutomationRegistryLogicB2_3 is AutomationRegistryBase2_3, Chainable {
   // |                      UPKEEP MANAGEMENT                       |
   // ================================================================
 
+  /**
+   * @notice adds fund to an upkeep
+   * @param id the upkeepID
+   * @param amount the amount of funds to add, in the upkeep's billing token
+   */
+  function addFunds(uint256 id, uint96 amount) external payable {
+    Upkeep memory upkeep = s_upkeep[id];
+    if (upkeep.maxValidBlocknumber != UINT32_MAX) revert UpkeepCancelled();
```

Comment:
oh man sad to see this one move to B :( that means 6000 extra gas every time users want to fund their upkeep. Hopefully we can improve this in a future version!